common/authorization/interceptor.go

```diff
@@ -133,6 +134,10 @@ func (a *interceptor) Interceptor(
 		}
 		if result.Decision != DecisionAllow {
 			scope.IncCounter(metrics.ServiceErrUnauthorizedCounter)
+			// if a reason is included in the result, include it in the error message
+			if result.Reason != "" {
+				return nil, fmt.Errorf("%w %s", errUnauthorized, result.Reason)
```

Comment:
I think it might be better to also add a `Reason` field to `serviceError.PermissionDenied` and return an instance of it instead or wrapping it in another error.

Reply:
I agree. It is bigger multi-repo PR though. `PermissionDenied` is here: https://github.com/temporalio/api-go/blob/master/serviceerror/permissionDenied.go. There are few other errors which has extra fields. You can just copy the style from there. Or I can do it for you.

Reply:
Okie doke. I've started the first PR: https://github.com/temporalio/api/pull/111

I guess once that's merged, I can update api-go, make a PR and then bring all that in here once merged, right?

Reply:
👋 Back at it. The next PR is reason for 👀 https://github.com/temporalio/api-go/pull/82

Reply:
It is merged. Please run `make update-proto` and it should get both protos and api-go version.